### DISCUSSION TITLE
I have two windows splitting the page, but they don't automatically resize together when I resize in the middle. Is this possible? :)

### DISCUSSION THREAD

**Discussion by JohnnyBonk** [rxhanson/Rectangle#1296] [Q&A]
I recently bought a mac which I am happy with and especially because of this rectangle app, but I'm missing a very important feature which is the ability to split screen in half and for example if i want the right side to take up more screen space, i would hope the left window adjusts when i drag the right out more, but instead it just covers up the left side instead of resizing together. is there a way to fix this, is this part of rectangle pro, or can we add code for this functionality?

**Comment by rxhanson** [comment]
Thanks for the feedback.

This is a part of Rectangle Pro, and it doesn't function identical to windows - the resize of the adjacent window will happen on mouse up, rather than mouse dragged, so I recommend trying it out and making sure you like it. The checkbox for this is "Resize adjacent windows after dragging window edge" in the general settings tab of Rectangle pro. There's also a checkbox on the window throw tab for resizing the adjacent window when you resize with the modifier keys + mouse movement configured there.

I don't have plans to bring this to Rectangle at this time (sorry!).

**Reply by JohnnyBonk** [reply]
Hi Ryan, Thank you for your quick response, I've downloaded the rectangle pro app and I'm still trying to figure out how to do that mouse up function. I am using a trackpad. I would be willing to donate to have that familiar feature.

**Reply by rxhanson** [reply]
By "mouse up" I mean when you release the mouse click, rather than while it is held down, if that helps clarify.

There is no upgrade path within the app, you'll have to download Rectangle Pro from https://rectangleapp.com/pro. Rectangle Pro is a separate app and is closed source. It has a lot more features, and therefore more complexity to figure out, and that's a large part why I have always kept them separate.

**Reply by JohnnyBonk** [reply]
Ah I see now I just figured it out. It will just take some getting used to. Thanks for your help and quick response time, i will buy the pro version.

**Reply by JohnnyBonk** [reply]
Only thing is sometimes it doesn't work. for example i ctrl command the two windows i want to both sides of the screen and then when i try to resize it doesn't work. maybe im missing something.

**Reply by JohnnyBonk** [reply]
Any guidance on how to get the resize to work on rectangle pro, even with mouse up most times it doesnt work.

**Reply by rxhanson** [reply]
Sorry for the delayed response. I have had a couple of users report that it doesn't always work, although I haven't been able to reproduce it (I use this a lot and it always works for me). There are a lot of variables at work here, so it's not the simplest problem to solve. Would you mind sending me the logs for a scenario where it doesn't work? In the menu bar menu, hold down the option key and select View Logging...

(If you did purchase Rectangle Pro and this is an immediate dealbreaker for you then just send me an email with any of your purchase information and I don't mind refunding)

**Reply by JohnnyBonk** [reply]
Hi Ryan, Im still having issues and cant figure out how to log scenario.

**Reply by rxhanson** [reply]
![image](https://github.com/rxhanson/Rectangle/assets/13651296/c281c4f0-5083-42d3-bf67-459fb0ce463f)

(holding option key)

Wow that's a large image.

With the log window open, try doing a resize and see if there's anything in the log viewer. If you don't mind, a screen recording might also help.

**Comment by neldeles** [comment]
Hi @rxhanson I noticed this feature doesn't work with [JankyBorders](https://github.com/FelixKratz/JankyBorders) enabled. If it's a simple fix to get it working with JankyBorders that'd be amazing! 

EDIT: Tried it with HazeOver and it doesn't work with that either. So it seems the feature isn't compatible with any app that applies a "focused" layer to apps.
